core/src/main/scala/org/http4s/EntityDecoder.scala

```diff
@@ -177,4 +177,11 @@ trait EntityDecoderInstances {
 
   implicit def multipart: EntityDecoder[Multipart] =
     MultipartDecoder.decoder
+
+  /** An entity decoder that ignores the content and returns unit. */
+  implicit val void: EntityDecoder[Unit] = new EntityDecoder {
+    val consumes: Set[MediaRange] = Set.empty
+
+    def decode(msg: Message, strict: Boolean): DecodeResult[Unit] = DecodeResult.success(())
```

Comment:
Most (all?) decoders consume the body, which is important for use in contexts like `client.toHttpService`.  Maybe we'd be better off with `DecodeResult.success(msg.body.kill.run)` here?